core/src/test/java/org/infinispan/stream/BaseStreamTest.java

```diff
@@ -419,8 +423,7 @@ public void testObjMin() {
       CacheSet<Map.Entry<Integer, String>> entrySet = cache.entrySet();
 
       assertEquals(Integer.valueOf(0),
-              createStream(entrySet).min((Serializable & Comparator<Map.Entry<Integer, String>>)
-                      (e1, e2) -> Integer.compare(e1.getKey(), e2.getKey())).get().getKey());
+              createStream(entrySet).min((e1, e2) -> Integer.compare(e1.getKey(), e2.getKey())).get().getKey());
```

Comment:
I wonder if the comparator would be serializable if you used `min(Comparator.comparing(Map.Entry::getKey())`...